src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useRef, useEffect } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  id: number;
+  completed: boolean;
+  title: string;
+  onItemDelete: (id: number) => void;
+  onItemStatusUpdate: (id: number) => void;
+  onItemTitleUpdate: (
```

Comment:
you can simplify this function (see other comments)
Also, write correct names for arguments